boa/precompile.py

```diff
@@ -0,0 +1,122 @@
+from typing import Optional
+from eth.codecs import abi
+from vyper.address_space import MEMORY
+from vyper.ast import parse_to_ast
+from vyper.codegen.core import IRnode
+from vyper.semantics.types.user import type_from_abi
+from vyper.semantics.types.utils import type_from_annotation
+from vyper.utils import binascii
+from boa.environment import register_precompile
+from vyper.builtins.functions import DISPATCH_TABLE, abi_encode, add_ofst, ir_tuple_from_args, process_inputs
+from vyper.builtins._signatures import BuiltinFunction
+from vyper.codegen.core import IRnode
+from vyper.semantics.types.shortcuts import (
+    UINT256_T,
+)
+
+def extract_arg_types(signature: str):
+    # Extract argument types and their names using regex
+
+    args = signature.split("(")[1].split(")")[0].split(",")
+    typs = [x.split(":")[1].strip().split("[")[0].lower() for x in args]
+    names = [x.split(":")[0].strip().split("[")[0].lower() for x in args]
+
+    # Combine the cleaned argument types and return them in parentheses
+    return f"({','.join(typs)})", names
+
+def parse_address_string(address: str | bytes) -> bytes:
+    # Convert address string to bytes
+    if isinstance(address, str):
+        if address.startswith("0x"):
+            address = address[2:]
+        return bytes.fromhex(address)
+    elif isinstance(address, bytes):
+        return binascii.unhexlify(bytes(address.hex(), "utf-8").zfill(40))
+
+class PrecompileBuiltin(BuiltinFunction):
+    # id can be name
+    _id = ""
+    # list of inputs (parsed from signature)
+    _inputs = []
+    # return type (parsed from signature)
+    _return_type = None
+    _address = None
```

Comment:
since these are set in the constructor, i don't think there is need for them to be listed out here